### PR TITLE
Added missing crud methods

### PR DESCRIPTION
**Pull request by oemel09**

### CODE REVIEW

File: src/main/java/com/ioet/bpm/skills/boundaries/SkillController.java [hunk 1]
```diff
@@ -22,9 +25,36 @@ public SkillController(SkillRepository skillRepository) {
         return new ResponseEntity<>(skills, HttpStatus.OK);
     }
 
+    @GetMapping("/{id}")
+    public Skill getSkill(@PathVariable(value = "id") String skillId) {
+        return skillRepository.findById(skillId)
+                .orElseThrow(() -> new SkillNotFoundException("Skill", "id", skillId));
+    }
```
Comment (enriquezrene) [inline src/main/java/com/ioet/bpm/skills/boundaries/SkillController.java]: This is not an error, just a record does not exist. A 404 is appropriate here.

File: src/main/java/com/ioet/bpm/skills/boundaries/SkillController.java [hunk 2]
```diff
@@ -22,9 +25,36 @@ public SkillController(SkillRepository skillRepository) {
...
     @PostMapping
     public ResponseEntity<Skill> create(@RequestBody Skill skill) {
         Skill skillCreated = skillRepository.save(skill);
         return new ResponseEntity<>(skillCreated, HttpStatus.CREATED);
     }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<?> delete(@PathVariable(value = "id") String skillId) {
+        Skill skill = skillRepository.findById(skillId)
+                .orElseThrow(() -> new SkillNotFoundException("Skill", "id", skillId));
```
Comment (enriquezrene) [inline src/main/java/com/ioet/bpm/skills/boundaries/SkillController.java]: same thing

File: src/main/java/com/ioet/bpm/skills/boundaries/SkillController.java [hunk 3]
```diff
@@ -22,9 +25,36 @@ public SkillController(SkillRepository skillRepository) {
...
+
+        skillRepository.delete(skill);
+
+        return ResponseEntity.ok().build();
+    }
+
+    @PutMapping("/{id}")
+    public Skill updateSkill(@PathVariable(value = "id") String skillId, @Valid @RequestBody Skill skillDetails) {
+        Skill skill = skillRepository.findById(skillId)
+                .orElseThrow(() -> new SkillNotFoundException("Skill", "id", skillId));
```
Comment (enriquezrene) [inline src/main/java/com/ioet/bpm/skills/boundaries/SkillController.java]: same here

File: src/main/java/com/ioet/bpm/skills/exception/SkillNotFoundException.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package com.ioet.bpm.skills.exception;
+
+import lombok.Getter;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@Getter
+@ResponseStatus(value = HttpStatus.NOT_FOUND)
+public class SkillNotFoundException extends RuntimeException {
```
Comment (enriquezrene) [inline src/main/java/com/ioet/bpm/skills/exception/SkillNotFoundException.java]: So, you can get rid of this class.

**Comment by oemel09** [comment]
I changed it so it just returns the 404

File: src/main/java/com/ioet/bpm/skills/boundaries/SkillController.java [hunk 5]
```diff
@@ -17,14 +20,49 @@ public SkillController(SkillRepository skillRepository) {
...
-    public ResponseEntity<Skill> create(@RequestBody Skill skill) {
+    public ResponseEntity<Skill> createSkill(@RequestBody Skill skill) {
         Skill skillCreated = skillRepository.save(skill);
         return new ResponseEntity<>(skillCreated, HttpStatus.CREATED);
     }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<?> deleteSkill(@PathVariable(value = "id") String skillId) {
+        Optional<Skill> skill = skillRepository.findById(skillId);
+        if (!skill.isPresent()) {
```
Comment (enriquezrene) [inline src/main/java/com/ioet/bpm/skills/boundaries/SkillController.java]: change the if with a positive condition

**Pull request merged**